src/dymaptic.GeoBlazor.Core/Scripts/arcGisJsInterop.ts

```diff
@@ -969,7 +970,44 @@ export async function addWidget(widget: any, viewId: string): Promise<void> {
                 if (hasValue(widget.label)) {
                     layerListWidget.label = widget.label;
                 }
+                break;
+            case 'basemapLayerList':
+                const basemapLayerListWidget = new BasemapLayerList({
+                    view: view
+                });
+                newWidget = basemapLayerListWidget;
                 
+                if (widget.HasCustomBaseListHandler) {
+                    basemapLayerListWidget.baseListItemCreatedFunction = async (evt) => {
+                        let dotNetBaseListItem = buildDotNetListItem(evt.item);
+                        let returnItem = await widget.layerListWidgetObjectReference.invokeMethodAsync('OnBaseListItemCreated', dotNetBaseListItem) as DotNetBaseListItem;
+                        evt.item.title = returnItem.title;
+                        evt.item.visible = returnItem.visible;
+                        // basemap will require additional implementation (similar to layerlist above) to activate additional layer and action sections.
+                        // evt.item.layer = returnItem.layer; --> needs implementation
+                        // evt.item.children = returnItem.children; --> needs implementation
+                        //evt.item.actionSections = returnItem.actionSections as any;
+                    };
+                }
+                if (widget.HasCustomReferenceListHandler) {
+                    basemapLayerListWidget.baseListItemCreatedFunction = async (evt) => {
+                        let dotNetReferenceListItem = buildDotNetListItem(evt.item);
+                        let returnItem = await widget.layerListWidgetObjectReference.invokeMethodAsync('OnReferenceListItemCreated', dotNetReferenceListItem) as DotNetReferenceListItem;
+                        evt.item.title = returnItem.title;
+                        evt.item.visible = returnItem.visible;
+                        // basemap will require additional implementation (similar to layerlist above) to activate additional layer and action sections.
+                        // evt.item.layer = returnItem.layer; --> needs implementation
+                        // evt.item.children = returnItem.children; --> needs implementation
+                        //evt.item.actionSections = returnItem.actionSections as any;
+                    };
+                }
+
+                if (widget.iconClass !== undefined && widget.iconClass !== null) {
```

Comment:
A few weeks ago I added the helper function `hasValue()` which can replace the long double-checks for undefined and null. See other widgets for example.